src/Middleware.php

```diff
@@ -3,31 +3,127 @@
 namespace Inertia;
 
 use Closure;
-use Illuminate\Support\Facades\Response;
-use Symfony\Component\HttpFoundation\RedirectResponse as Redirect;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
 
-class Middleware
+abstract class Middleware
 {
-    public function handle($request, Closure $next)
+    /**
+     * Determines the current Inertia asset version hash.
+     *
+     * @see https://inertiajs.com/asset-versioning
+     * @param  \Illuminate\Http\Request  $request
+     * @return string|null
+     */
+    public function version(Request $request)
     {
+        //
```

Comment:
Also, I was thinking.. what if we would do this in the _base_ Middleware?:

```suggestion
        // When we are running on Laravel Vapor, asset URLs are automatically updated
        // when the application is being deployed. Because of this, we do not need
        // to use any files for hashing, as we can simply use this URL instead.
        if (config('app.asset_url')) {
            return md5(config('app.asset_url'));
        }

        // Alternatively, when we are running in a regular (non-serverless) environment
        // we'll attempt to use the Laravel Mix asset manifest to generate our hash.
        if (file_exists($manifest = public_path('mix-manifest.json'))) {
            return md5_file($manifest);
        }
```

Since this adapter is Laravel specific to begin with, this would allow users to enable it using an one-liner, and would also skip the need to add this directly into Jetstream.

In addition, it still won't be enabled by default, because the `inertia:middleware` command generates the Middleware without the `parent::version($request)` call

Reply:
Honestly, I'm tempted to just do this, and have it enabled by default (not commented out in the generated middleware), and folks can always change it if they need to.

This seems like a totally reasonable way of determining the asset version in an automated way.